### DISCUSSION TITLE
add support for controlnet

### DISCUSSION THREAD

**Discussion by x-legion** [Coyote-A/ultimate-upscale-for-automatic1111#45] [Ideas]

**Comment by WalternateBishop** [comment]
While this won't be the holy grail of upscaling, it'll be incredibly close. Can't wait to see this implemented!

**Comment by Preston-L** [comment]
This wouldn't just be useful for upscaling. you would be able to do amazing style transformations while preserving intricate details!

**Comment by Lute-4** [comment]
+1 :)

https://github.com/Mikubill/sd-webui-controlnet/issues/43

**Comment by Lute-4** [comment]
Maybe useful, "added an API layer":
https://github.com/Mikubill/sd-webui-controlnet/pull/194
